proseco/guide.py

```diff
@@ -38,66 +37,20 @@ def get_guide_catalog(obsid=0, att=None, date=None, t_ccd=None, dither=None, n_g
     :returns: GuideTable of acquisition stars
     """
 
-    guides = GuideTable(print_log=print_log)
+    guides = GuideTable()
+    guides.set_kwargs(obsid=obsid, **kwargs)
+    guides.set_stars()
 
-    # If an explicit obsid is not provided to all getting parameters via mica
-    # then all other params must be supplied.
-    all_pars = all(x is not None for x in (att, t_ccd, date, dither))
-    if obsid == 0 and not all_pars:
-        raise ValueError('if `obsid` not supplied then all other params are required')
-
-    # If not all params supplied then get via mica for the obsid.
-    if not all_pars:
-        from mica.starcheck import get_starcheck_catalog
-        guides.log(f'getting starcheck catalog for obsid {obsid}')
-
-        obs = get_starcheck_catalog(obsid)
-        obso = obs['obs']
-
-        if att is None:
-            att = [obso['point_ra'], obso['point_dec'], obso['point_roll']]
-        if date is None:
-            date = obso['mp_starcat_time']
-        if t_ccd is None:
-            t_ccd = obs.get('pred_temp') or -15
-
-        dither_y_amp = obso.get('dither_y_amp')
-        dither_z_amp = obso.get('dither_z_amp')
-        if dither_y_amp is not None and dither_z_amp is not None:
-            dither = ACABox((dither_y_amp, dither_z_amp))
-
-        if n_guide is None:
-            cat = obs['cat']
-            n_guide = 8 - np.count_nonzero((cat['type'] == 'FID') | (cat['type'] == 'MON'))
-
-    if not isinstance(dither, ACABox):
-        dither = ACABox(dither)
-
-    if dark is None:
+    if guides.dark is None:
         from mica.archive.aca_dark import get_dark_cal_image
-        guides.log(f'getting dark cal image at date={date} t_ccd={t_ccd:.1f}')
-        dark = get_dark_cal_image(date=date, select='before', t_ccd_ref=t_ccd, aca_image=True)
+        guides.log(f'getting dark cal image at date={guides.date} t_ccd={guides.t_ccd:.1f}')
+        guides.dark = get_dark_cal_image(date=guides.date, select='before',
```

Comment:
Doesn't matter at all for operational use but we've got a 'before' here and a 'nearest' in acq.

Reply:
Fixed.  (It was in report.py not acq.py though).